File: 1876-substrings-of-size-three-with-distinct-characters/1876-substrings-of-size-three-with-distinct-characters.js
var countGoodSubstrings = function (s) {
  let res = 0;
  for (let i = 0; i < s.length - 2; i++) {
    const window = [s[i], s[i + 1], s[i + 2]];
    const chars = new Set(window);

    if (chars.size === 3) {
      res++;
    }
  }
  return res;
};

File: 1876-substrings-of-size-three-with-distinct-characters/1876-substrings-of-size-three-with-distinct-characters.cpp
class Solution
{
public:
  int countGoodSubstrings(string s)
  {
    int res = 0;

    unordered_map<char, int> map{};
    for (int i = 0; i < s.size(); i++)
    {
      if (i > 2 and --map[s[i - 3]] == 0)
      {
        map.erase(s[i - 3]);
      }
      map[s[i]]++;
      if (map.size() == 3)
        res++;
    }
    return res;
  }
};